subcell_pipeline/analysis/compression_metrics/vectors.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any, Tuple, Union
+
+import numpy as np
+
+from subcell_pipeline.analysis.compression_metrics.constants import ABSOLUTE_TOLERANCE
+
+
+def get_unit_vector(
+    vector: np.array,
+) -> Tuple[np.array, Union[float, np.floating[Any]]]:
+    """
+    Calculate the unit vector and length of a given vector.
+
+    Parameters
+    ----------
+    vector
+        The input vector to calculate unit vector and length.
+
+    Returns
+    -------
+    :
+        A tuple containing the unit vector and length of the input vector.
+    """
+    if np.linalg.norm(vector) < ABSOLUTE_TOLERANCE or np.isnan(vector).any():
+        return np.array([0, 0, 0]), 0.0
+    else:
+        vec_length = np.linalg.norm(vector)
+        return vector / vec_length, vec_length
+
+
+def get_end_to_end_unit_vector(
+    polymer_trace: np.ndarray,
+) -> Tuple[np.array, Union[float, np.floating[Any]]]:
```

Comment:
If possible, each method should only return one thing. Can you have a separate method handle vector lengths?

Reply:
Fixed in [3dd2d6](https://github.com/simularium/subcell-pipeline/pull/58/commits/3dd2d6e1d02df67ab33d08977268c6dfeb5911f3)